ironfish/src/rpc/routes/chain/followChain.ts

```diff
@@ -74,35 +112,60 @@ router.register<typeof FollowChainStreamRequestSchema, FollowChainStreamResponse
       head: head,
     })
 
-    const send = (header: BlockHeader, type: 'connected' | 'disconnected' | 'fork') => {
+    const send = async (block: Block, type: 'connected' | 'disconnected' | 'fork') => {
+      const transactions = await Promise.all(
+        block.transactions.map(async (transaction) => {
+          const size = Buffer.from(
+            JSON.stringify(node.strategy.transactionSerde.serialize(transaction)),
+          ).byteLength
+
+          return {
+            hash: BlockHashSerdeInstance.serialize(transaction.transactionHash()),
+            size,
+            fee: Number(await transaction.transactionFee()),
+            notes: [...transaction.notes()].map((note) => ({
+              commitment: note.merkleHash().toString('hex'),
+            })),
+            spends: [...transaction.spends()].map((spend) => ({
+              nullifier: spend.nullifier.toString('hex'),
+            })),
+          }
+        }),
+      )
+
       request.stream({
         type: type,
         head: {
           sequence: node.chain.head.sequence,
         },
         block: {
-          hash: header.hash.toString('hex'),
-          sequence: header.sequence,
-          previous: header.previousBlockHash.toString('hex'),
-          graffiti: GraffitiUtils.toHuman(header.graffiti),
-          work: header.work.toString(),
+          hash: block.header.hash.toString('hex'),
+          sequence: block.header.sequence,
+          previous: block.header.previousBlockHash.toString('hex'),
+          graffiti: GraffitiUtils.toHuman(block.header.graffiti),
+          work: block.header.work.toString(),
           main: type === 'connected',
-          timestamp: header.timestamp.valueOf(),
-          difficulty: header.target.toDifficulty().toString(),
+          timestamp: block.header.timestamp.valueOf(),
+          difficulty: block.header.target.toDifficulty().toString(),
+          transactions,
         },
       })
     }
 
-    const onAdd = (header: BlockHeader) => {
-      send(header, 'connected')
+    const onAdd = async (header: BlockHeader) => {
+      const block = await node.chain.getBlock(header)
+      Assert.isNotNull(block)
+      void send(block, 'connected')
     }
 
-    const onRemove = (header: BlockHeader) => {
-      send(header, 'disconnected')
+    const onRemove = async (header: BlockHeader) => {
+      const block = await node.chain.getBlock(header)
+      Assert.isNotNull(block)
+      void send(block, 'disconnected')
     }
 
     const onFork = (block: Block) => {
-      send(block.header, 'fork')
+      void send(block, 'fork')
```

Comment:
This could cause problems but we can just see what happens for now if we don't await this.

Reply:
Changed to `await` and everything continues to work as expected.